components/camera/replaypcd/replaypcd_test.go

```diff
@@ -2,43 +2,397 @@ package replaypcd
+			Source: "test",
+			Interval: TimeInterval{
+				End: "2000-01-01T12:00:10Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		endTime, err := time.Parse(timeFormat, replayCamCfg.Interval.End)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through files that meet the provided filter
+		for i := 0; i < endTime.Second(); i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with start filter", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:05Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		startTime, err := time.Parse(timeFormat, replayCamCfg.Interval.Start)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through files that meet the provided filter
+		for i := startTime.Second(); i < numPCDFiles; i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with filter no data", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:30Z",
+				End:   "2000-01-01T12:00:40Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
```

Comment:
Can we add a test showing that the consumer can still get data from the replay camera after the replay camera has returned an errEndOfDataset? We can hack our implementation of BinaryDataByFilter to know when it's reach the end of its dataset, and then have an extra file (maybe just one of the existing files) that it returns after that point. This would simulate a new file being "synced," and if anyone ever tried to change the replay camera to just return errEndOfDataset indefinitely after it had returned no data, this test would fail, indicating to the engineer that they have broken an intended feature.